wave_lang/kernel/wave/compile.py

```diff
@@ -215,6 +215,12 @@ def wave_compile(options: WaveCompileOptions, kernel: "LaunchableWave") -> WaveK
     cache_manager = None
     binary_path = None
 
+    # Since constraints are used to lookup the compiled kernel in the cache,
+    # we initialize/update the constraints _before_ the cache lookup.
+    kernel.initialize_wave_constraints()
```

Comment:
Why move these from the pass pipeline to here? Aren't the constraints always guaranteed to be initialized by the time we arrive here?

Reply:
They aren't.  Without this change, these calls (and thus the constraint initialization) happen after the cache lookup, not before, and this patch instead moves these calls to happen before the cache lookup.

Reply:
makes sense, thanks!